CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ CHANGELOG
   - Intel Python: 2023.2.0
   - Critical Update for Intel oneAPI DPC++/C++ Compiler: 2023.2.1
   - Critical Update for Intel Fortran Compiler & Intel® Fortran Compiler Classic: 2023.2.1
+  - Remove `CommunicationParameters` from the Custom Slurm Settings deny list
```

Comment:
Why isn't there an entry in the CLI changelog for the upgrade of Slurm? I will approve this PR as it is, but I'm fine with adding the changelog entry within this PR.

Reply:
I assumed it was already there from when we actually upgraded to slurm 23.11.

I'll add it with this PR.